src/python/pants/engine/internals/engine_test.py

```diff
@@ -314,40 +313,39 @@ def _fixture_for_rules(
             max_workunit_verbosity=max_workunit_verbosity,
             specs=Specs.empty(),
             options_bootstrapper=create_options_bootstrapper([]),
+            pantsd=False,
         )
         return scheduler, tracker, handler
 
     def test_streaming_workunits_reporting(self):
         scheduler, tracker, handler = self._fixture_for_rules([fib, QueryRule(Fib, (int,))])
-        with handler.session(pantsd=False):
+        with handler:
             scheduler.product_request(Fib, subjects=[0])
-
         flattened = list(itertools.chain.from_iterable(tracker.finished_workunit_chunks))
         # The execution of the single named @rule "fib" should be providing this one workunit.
-        self.assertEqual(len(flattened), 1)
+        assert len(flattened) == 1
 
-        tracker.finished_workunit_chunks = []
-        with handler.session(pantsd=False):
+        scheduler, tracker, handler = self._fixture_for_rules([fib, QueryRule(Fib, (int,))])
+        with handler:
             scheduler.product_request(Fib, subjects=[10])
 
-        # Requesting a bigger fibonacci number will result in more rule executions and thus more reported workunits.
-        # In this case, we expect 10 invocations of the `fib` rule.
+        # Requesting a bigger fibonacci number will result in more rule executions and thus
+        # more reported workunits. In this case, we expect 11 invocations of the `fib` rule.
         flattened = list(itertools.chain.from_iterable(tracker.finished_workunit_chunks))
-        assert len(flattened) == 10
+        assert len(flattened) == 11
```

Comment:
This is now 11 because we don't have the memoization from the original scheduler. We created a new one.